src/histolab/slide.py

```diff
@@ -73,29 +117,43 @@ def dimensions(self) -> Tuple[int, int]:
         dimensions : Tuple[int, int]
             Slide dimensions (width, height)
         """
+        if self._use_largeimage:
+            return self._metadata["sizeX"], self._metadata["sizeY"]
+
         return self._wsi.dimensions
 
     def extract_tile(
-        self, coords: CoordinatePair, level: int, tile_size: Tuple
+        self,
+        coords: CoordinatePair,
+        tile_size: Tuple,
+        level: int = None,
+        mpp: float = None,
     ) -> Tile:
         """Extract a tile of the image at the selected level.
 
         Parameters
         ----------
         coords : CoordinatePair
             Coordinates at level 0 from which to extract the tile.
+        tile_size: tuple
+            Final size of the extracted tile (x,y).
         level : int
             Level from which to extract the tile.
-        tile_size: tuple
-            Final size of the tile (x,y).
+        mpp : float
+            Micron per pixel resolution. Takes precedence over level.
 
         Returns
         -------
         tile : Tile
             Image containing the selected tile.
         """
+        assert (level is not None) or (
+            mpp is not None
+        ), "either level or mpp must be provided!"
```

Comment:
I would change it with a raise exception but also use the inverted logic:
```
If level is None and mpp is none:
    raise ....
```
I think using the `and` is more explicit